src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -227,6 +229,21 @@ public HubConnection(IConnectionFactory connectionFactory, IHubProtocol protocol
         _state = new ReconnectingConnectionState(_logger);
 
         _logScope = new ConnectionLogScope();
+
+        var options = serviceProvider.GetService<IOptions<HubConnectionOptions>>();
+
+        if (options != null)
```

Comment:
nit: We can change this section of code into 2 lines by using nullable operators and the default values
e.g. `ServerTimeout = options?.Value.ServerTimeout ?? DefaultServerTimeout;`